llvm/lib/SandboxIR/Tracker.cpp

```diff
@@ -160,26 +160,28 @@ void CallBrInstSetIndirectDest::dump() const {
 }
 #endif
 
-SetVolatile::SetVolatile(Instruction *I, bool WasBool, Tracker &Tracker)
-    : IRChangeBase(Tracker), Inst(I) {
-  if (auto *Load = cast<llvm::LoadInst>(Inst)) {
+SetVolatile::SetVolatile(Instruction *I, Tracker &Tracker)
+    : IRChangeBase(Tracker) {
+  if (auto *Load = cast<llvm::LoadInst>(I)) {
```

Comment:
This should be a `dyn_cast<LoadInst>(I)`. A `cast` will assert that the type is correct, so this will crash. Also `I` is a `sandboxir::LoadInst` or `sandboxir::StoreInst`, so we should cast it to `<LoadInst>` not `<llvm::LoadInst>`.

Reply:
there's no viable overloaded '='. So when i remove  `llvm::` from `llvm::LoadInst`, making `InstUnion = Load` incorrect. So I end up getting compiler errors. Should I implement the operator overloads for StoreInst and LoadInst or is there another solution?

Reply:
This can't be right. You may have missed a `*` somewhere.
There is very similar code in `RemoveFromParent::revert` constructor. In that class `NextInstrOrBB` is a `PointerUnion<Instruction *, BasicBlock *>` and we can cast it as `if (auto *NextI = NextInstrOrBB.dyn_cast<Instruction *>())` where `auto` is `Instruction`.